### ISSUE TITLE
Create Table: permission_type

### ISSUE THREAD

**Issue by Neecolaa** [hackforla/peopledepot#24]
### Dependency
- #147 

### Overview

We need to create the permission_type table so that we can update a shared data store across hackforla.org, vrms, civictechjobs, and tables (onboarding) project. 

#### Details

A table and a model are the same thing

### Action Items

- [x] identify and document table description (see spreadsheet under Resources)
  - [x] #176
  - Long Description: i.e., adminVrms, adminBrigade, adminProject, member
  - Sheet: PermissionType - Data

id | name | description
-- | -- | --
  | adminGlobal | Can CRUD anything
  | adminVrms | Can R anything and update permissions for users to the level of Admin Brigade
  | adminBrigade | Can CRUD anything related to a user or project assigned to their brigade
  | adminProject | Can Read and Update anything related to their assigned project
  | memberProject | Can Read anything related to their project that is visible
  | memberGeneral | Can express interest in a project's open role if they are the role

- [x] compare and check off the data fields (below) against the ERD. Note any fields in the ERD that's missing from the list. (see Resources)
- [x] compare the check off associated tables (below) against the ERD. Note any associated tables in the ERD that's missing from the list (see Resources)
  - [ ] `permission` table is a related table and is not yet created
	- [ ] #22 seems to require some further investigation
	- [ ] as mentioned in #147, there seems to be a need to further investigate & get clarity on how to implement permissions
  - [ ] `permission_history` table is a related table and is not yet created
	- [ ] #37 will need to investigate further later
- [x] create a single model in Django (defining schema)
- [ ] add seed data
- [ ] write a test for the relationships this model will have with other models (e.g., creating a user and assigning them a set of permissions on a project).
- [x] write an API end point  
![image](https://github.com/hackforla/peopledepot/assets/86270711/91b131b2-1808-4682-bd06-ff98467b8955)

- [x] write API unit tests
  - [x] To resume at [Adding API Tests](https://github.com/hackforla/peopledepot/blob/main/docs/how-to/add-model-and-api-endpoints.md#add-api-tests)
- [ ] document the endpoint

### Resources/Instructions

- See [People Depot Resources wiki page](https://github.com/hackforla/peopledepot/wiki/Resources-and-Links) for links
  - ERD
  - Table and Field Definitions
  - API Endpoint
- See the [wiki page](https://github.com/hackforla/peopledepot/wiki/Create-Table-issues-data-gathering-workflow) for instructions.

### Description

Dictionary of permission types used in #22 

### Data Fields

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] (PK) id - int - Record Id
   - [x] name - varchar
   - [x] description - varchar

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

### Associated Tables

1. Copied from spreadsheet and checked off according to ERD. (unchecked items indicate a mismatch between ERD and spreadsheet, which requires a review)
   - [x] permission (one-to-many)
   - [x] permission_history (one-to-many)

1. In ERD only (having items here indicates a mismatch, which requires a review)
   - None

**Comment by Neecolaa** [comment]
Currently, this table isn't storing what each permission allows, nor is that information being stored somewhere else. Do we want to store the CRUD information here? Could use booleans (can_create, can_read, can_update, can_delete), it would depend on how we want to set up permissions

**Comment by fyliu** [comment]
I just realized this is the same kind of issue as #35 and should be implemented the same way. I added a dependency to this and will release this issue.

**Comment by yoyoyojoe** [comment]
Got clarification, will remove dependency, and will try to work on this issue.

> ### Dependency 
> - #35

**Comment by fyliu** [comment]
Hmm... this issue should have been put into the ice box for a dependency not addressed above. 

- See #147

This is still okay to work on although not ideal. We can always create an issue to modify this model later.

**Comment by yoyoyojoe** [comment]
Handling permissions definitely is complex and complicated. I'd love to investigate more on how everything relates to each other to get clarity. Might be hard with the new job, but I'll try to be at the meeting tomorrow a bit after 5pm.

**Comment by ExperimentsInHonesty** [comment]
@yoyoyojoe fang says that this issue is dependent on #147.  So we recommend putting this issue back into the icebox. If you need help picking a new issue, let us know.

**Comment by fyliu** [comment]
I'm unassigning @joey-ma in case anyone wants to work on this.

Be aware that Joey has done previous work on this in draft PR #158. Anyone working on this should try to preserve the work that's already done by checking out that branch and continuing from there rather than start from scratch.

**Comment by ExperimentsInHonesty** [comment]
@alexlaw528 
Please provide update
1. Progress: "What is the current status of your project? What have you completed and what is left to do?"
2. Blockers: "Difficulties or errors encountered."
3. Availability: "How much time will you have this week to work on this issue?"
4. ETA: "When do you expect this issue to be completed?"
5. Pictures or links* (if necessary): "Add any pictures or links that will help illustrate what you are working on."

* remember to add links to the top of the issue if they are going to be needed again.

**Comment by alexlaw528** [comment]
@ExperimentsInHonesty 
1. Progress - Still working on getting comfortable with Django. 
2. Blockers - None at the moment.
3. Availability - Will have 4-6 hrs this weekend to work on this.
4. ETA  - Hoping to complete by end of next weekend (9/24)

**Comment by alexlaw528** [comment]
@fyliu - can you review PR #201?

Some of the other requests in this issue will likely require updating to include custom permission. But that seems out of scope of this issue.

**Comment by fyliu** [comment]
Closing issue since the PR is merged.
